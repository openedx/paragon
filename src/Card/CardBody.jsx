import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const CardBody = ({ className, children }) => (
  <div className={classNames('pgn__card-body', className)}>
    {children}
  </div>
);

CardBody.propTypes = {
  /** Specifies the content of the component. */
  children: PropTypes.node,
  /** The class to append to the base element. */
  className: PropTypes.string,
};

CardBody.defaultProps = {
  children: undefined,
  className: undefined,
};

export default CardBody;
