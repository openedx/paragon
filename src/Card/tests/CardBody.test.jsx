import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import CardBody from '../CardBody';

describe('correct rendering', () => {
  it('renders CardBody element', () => {
    const tree = renderer.create((
      <CardBody>Text</CardBody>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct base className', () => {
    const body = mount(<CardBody />).find('div');

    expect(body.hasClass('pgn__card-body')).toBe(true);
  });

  it('renders body with custom className', () => {
    const className = 'my-class-name';
    const body = mount(<CardBody className={className} />).find('div');
    expect(body.hasClass('my-class-name')).toBe(true);
  });
});
