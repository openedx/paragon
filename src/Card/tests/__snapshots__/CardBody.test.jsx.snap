// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct rendering renders CardBody element 1`] = `
<div
  className="pgn__card-body"
>
  Text
</div>
`;
